AWSAppSyncClient/Internal/AuthInterceptor/LambdaAuthInterceptor.swift

```diff
@@ -61,6 +67,34 @@ class LambdaAuthInterceptor: AuthInterceptor {
         let signedRequest = AppSyncConnectionRequest(url: url)
         return signedRequest
     }
+    
+    private func retrieveLatestAuthToken() -> Swift.Result<String, Error> {
+        guard let asyncAuthProvider = authTokenProvider as? AWSLambdaAuthProviderAsync else {
+            return .success(authTokenProvider.getLatestAuthToken())
+        }
+        var authToken: String?
+        var authTokenError: Error?
+        let result: Swift.Result<String, Error>
+        
+        let semaphore = DispatchSemaphore(value: 0)
+        
+        asyncAuthProvider.getLatestAuthToken { (token, error) in
+            authToken = token
+            authTokenError = error
+            semaphore.signal()
+        }
+        semaphore.wait()
+        
+        if let authTokenError = authTokenError {
+            result = .failure(authTokenError)
+        } else if let authToken = authToken {
+            result = .success(authToken)
+        } else {
+            fatalError("Incompatible values for authorization token and error: nil, nil")
```

Comment:
is `assert` the right thing to do here? if the problem is just returning _something_, can return our own failure
```
assert(true, "Incompatible values for authorization token and error: nil, nil")
result = .failure(TokenError.unknown)
```